### PR TITLE
many glm fix

### PR DESCRIPTION
**Pull request by ferrine**
Fixed a bug when it is not possible to run more than one glm model in a model

### CODE REVIEW

**Review by springcoil — APPROVED** [review]
I don't see any problem with this. Do you @twiecki

**Comment by ferrine** [comment]
I also complain about difficult interface for glm. I had to write an ugly code to adopt a matrix for that function. In my opinion it should take not data and formula as arguments, but X and y (with optional data + formula). That probably changes the api so optional X+y can be better solution.
Lets make a class of it?

```
class glm(collections.namedtuple('glm', 'y_est,coeffs'):
    ...
```

File: pymc3/glm/glm.py [hunk 1]
```diff
@@ -1,23 +1,41 @@
...
+    # else -> pd.DataFrame -> ok
+    data = pd.concat([y, X], 1)
+    formula = patsy.ModelDesc(
+        [patsy.Term([patsy.LookupFactor(y.name)])],
+        [patsy.Term([patsy.LookupFactor(p)]) for p in X.columns]
+    )
+    return data, formula
+
+
+class linear_component(namedtuple('Estimate', 'y_est,coeffs')):
```
Comment (twiecki) [inline pymc3/glm/glm.py:38]: What does this do?

**Comment by twiecki** [comment]
I think this refactoring makes sense. The added code complexity is warranted by maintaining API compatibility but I do wonder if it's not better to change API, in case you have a good idea of how it could be better?

**Comment by ferrine** [comment]
@twiecki Better idea is implementing glm as a class. It will be more flexible in the future. Moreover linear component for tensors as well as matrices or DataFrame is needed too.

File: pymc3/glm/glm.py (hunk 1 above)
Comment (ferrine) [reply pymc3/glm/glm.py:38]: It is a tuple subclass with special constructor. Before you had a function that returned a tuple, now you have a tuple that can be constructed in two ways. Call result is still tuple and thus compatible with old API

**Comment by twiecki** [comment]
OK, are you interested in doing this? I'm not opposed to changing the API in the way you propose. Perhaps also move this to `pymc3.models` while at it. In any case, we can merge this to fix the problems it has now and improve it later.

**Pull request merged**